extensions/azurePublishNew/src/components/provisioningWizards/steps/ResourceConfigurationStep.tsx

```diff
@@ -0,0 +1,268 @@
+import { SubscriptionPicker } from '../../resourceConfiguration/SubscriptionPicker';
+import { TenantPicker } from '../../resourceConfiguration/TenantPicker';
+import { DeployLocationPicker } from '../../resourceConfiguration/DeployLocationPicker';
+
+type Props = {
+  onResourceConfigurationChange: (isValidConfiguration: boolean) => void;
+};
+
+const ConfigureResourcesSectionName = styled(Text)`
+  font-size: ${FluentTheme.fonts.mediumPlus.fontSize};
+  font-weight: ${FontWeights.semibold};
+  margin-bottom: 4px;
+`;
+
+const ConfigureResourcesSectionDescription = styled(Text)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  line-height: ${FontSizes.size14};
+  margin-bottom: 20px;
+`;
+
+const configureResourcePropertyStackTokens: IStackTokens = { childrenGap: 5 };
+
+const configureResourcePropertyLabelStackStyles: IStackItemStyles = {
+  root: {
+    width: '200px',
+  },
+};
+
+const ConfigureResourcesPropertyLabel = styled(Label)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
+  font-weight: ${FontWeights.regular};
+`;
+
+const autoCompleteTextFieldStyles = { root: { paddingBottom: '4px', width: '300px' } };
+
+const configureResourcesIconStyle = {
+  root: {
+    color: NeutralColors.gray160,
+    userSelect: 'none',
+  },
+};
+
+const LearnMoreLink = styled(Link)`
+  user-select: none;
+  font-size: 14px;
+`;
+
+const FullWidthForm = styled.form`
+  width: '100%';
+`;
+
+//TODO: a wrapper function would be nice eg: formatMessages({})
+const messages = {
+  azureDetails: formatMessage('Azure details'),
+  azureDirectory: formatMessage('Azure Directory'),
+  selectAzureDirectory: formatMessage('Select your Azure directory and subscription, enter resource group name.'),
+  learnMore: formatMessage('Learn more'),
+  enterResourceName: formatMessage('Enter resource name and select region. This will be applied to the new resources.'),
+  activeDirectoryInfo: formatMessage(
+    'Azure Active Directory is Microsoft’s cloud-based identity and access management service.'
+  ),
+  subscription: formatMessage('Subscription'),
+  subscriptionInfo: formatMessage('The subscription that will be billed for the resources.'),
+  resourceGroup: formatMessage('Resource group'),
+  resourceGroupInfo: formatMessage(
+    'A custom resource group name that you choose or create. Resource groups allow you to group Azure resources for access and management.'
+  ),
+  resourceDetails: formatMessage('Resource details'),
+  name: formatMessage('Name'),
+  uniqueResourceName: formatMessage('A unique name for your resources.'),
+  region: formatMessage('Region'),
+  luisRegion: formatMessage('LUIS region'),
+  regionIconInfo: formatMessage('The region where your resources and bot will be used.'),
+  luisRegionIconInfo: formatMessage('The region associated with your Language understanding model.'),
+};
+export const ResourceConfigurationStep = (props: Props) => {
+  const [userInfo, setUserInfo] = useRecoilState(userInfoState);
```

Comment:
have you considered encapsulating all these setters in a dispatcher state using a recoil callback? (similar to composer client)
Right now, your component is mutating a shared state directly.

Reply:
No, I haven't. I will take a look  at the composer implementation.